src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -121,11 +123,23 @@ public Integer getTotalUnReviewedDars(AuthUser authUser) {
 
     private List<DataAccessRequestManage> createAccessRequestManage(List<Document> documents, Map<String, Election> electionList) {
         List<DataAccessRequestManage> requestsManage = new ArrayList<>();
+        Map<Integer, Integer> datasetDacPairs = dataSetDAO.findDatasetAndDacIds().
+                stream().
+                collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
         documents.forEach(dar -> {
             DataAccessRequestManage darManage = new DataAccessRequestManage();
             ObjectId id = dar.get(DarConstants.ID, ObjectId.class);
-            List<Integer> dataSets = DarUtil.getIntegerList(dar, DarConstants.DATASET_ID);
-            List<DataSet> dataSetsToApprove = dataSetDAO.findNeedsApprovalDataSetByDataSetId(dataSets);
+            List<Integer> darDatasetIds = DarUtil.getIntegerList(dar, DarConstants.DATASET_ID);
+            List<DataSet> dataSetsToApprove = dataSetDAO.findNeedsApprovalDataSetByDataSetId(darDatasetIds);
+            if (darDatasetIds.size() > 1) {
+                darManage.getErrors().add("DAR has more than one dataset association: " + ArrayUtils.toString(darDatasetIds));
```

Comment:
Instead of calling a getter for the error list and added value. It's efficient to add a function to DataAccessRequestManage.java to add errors.